### PR TITLE
feat(alerts): Show alert to opt-in to metric alerts without archived issue events

### PR DESCRIPTION
**Pull request by scttcper**
- Adds a banner to alert details telling people they can filter their errors metric alerts to exclude archived issues.
- Banner takes them to the edit page with `is:unresolved` added to their query
- https://www.notion.so/sentry/Metric-alerts-Ignoring-Archived-Issues-8b5ce2b77a084256bda3591394372a14?pvs=4#51ce2e7fef7a483cb652818d7c5ca7a0
- The events-stats query (the chart) on the edit page should display the metric alert with `is:unresolved` and `dataset=errors` (which is different from the metric alert dataset - in this case is 'events') 🤷 
- can dismiss the alert for the user

On alert details

Screenshot 2023-11-21 at 11 19 55 AM

On edit, reusing the @getsentry/telemetry-experience migration experience to show only the threshold fields. The chart doesn't work yet since the `dataset=errors` query parameter is not yet supported.
Screenshot 2023-11-21 at 11 20 29 AM
reduced layout.

does not yet handle:
 - Adding `is:unresolved` to the default error alert

### CODE REVIEW

**Comment by 17hogeju** [comment]
Is there a Github issue link for this PR or can you provide a more detailed description?

File: static/app/views/alerts/rules/metric/details/body.tsx [hunk 1]
```diff
@@ -221,6 +225,26 @@ export default function MetricDetailsBody({
...
+                  to={migrationFormLink}
+                  size="xs"
+                  icon={<IconEdit size="xs" />}
+                >
+                  {t('Review Thresholds')}
+                </LinkButton>
+              }
+            >
+              {t(
+                'Sentry now supports excluding archived issues from metric alerts. Please review your alert thresholds.'
```
[the remark this answers is not in the document]
Comment (scttcper) [reply static/app/views/alerts/rules/metric/details/body.tsx:243]: @rachrwang maybe we can word this better.
image

**Comment by scttcper** [comment]
@17hogeju added more details

File: static/app/views/alerts/rules/metric/ruleForm.tsx [hunk 2]
```diff
@@ -998,6 +1010,10 @@ class RuleFormContainer extends DeprecatedAsyncComponent<Props, State> {
 
     const showMigrationWarning =
```
Comment (ArthurKnaus) [inline static/app/views/alerts/rules/metric/ruleForm.tsx:1011]: Let's also rename it here?
```suggestion
    const showTransactionMigrationWarning =
```

**Review by ArthurKnaus — APPROVED** [review]

**Pull request merged**